pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -643,12 +657,28 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
           java.sql.Timestamp tmpts;
           if (in instanceof java.util.Date) {
             tmpts = new java.sql.Timestamp(((java.util.Date) in).getTime());
+            //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+          } else if (in instanceof LocalDateTime) {
+            tmpts = java.sql.Timestamp.valueOf((LocalDateTime)in);
+            //#endif
           } else {
             tmpts = connection.getTimestampUtils().toTimestamp(null, in.toString());
           }
           setTimestamp(parameterIndex, tmpts);
         }
         break;
+      //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+      case Types.TIMESTAMP_WITH_TIMEZONE:
+        if (in instanceof OffsetDateTime) {
+          setTimestamp(parameterIndex, java.sql.Timestamp.from(((OffsetDateTime)in).toInstant()));
```

Comment:
I think this should be corrected.
You'd better use `setTimestamp(int, Timestamp, Calendar)` method.
Is there a reason to use the one with "default time zone" method?